docs/start/resourceconfig.md

```diff
@@ -0,0 +1,78 @@
+imposes a limit of 7g of memory. This can be used in conjunction with the following `cgrules.conf` [config file](https://linux.die.net/man/5/cgrules.conf):
+
+```
+agent_user:Agent.Listener memory agent
+agent_user:Agent.Worker memory agent
+agent_user memory job
+```
+It is important to use two groups, because otherwise the pipeline job processes will inherit the group from their parent, the agent, so there will be no distinction in terms of control.
+A second "job" cgroup allows the job processes to be managed independent of the agent, e.g. in an out-of-memory scenario (when the job exceeds the limits given by the `job` cgroup), the job
+will be killed instead of the agent. If a single cgroup is used, the agent may killed to reclaim memory from the cgroup.
+
+Additionally, using two groups can provide the agent "dedicated" memory to avoid instability caused by thrashing. In the above example `cgconfig.conf`, if the job group memory limit is 7g and the host machine
+has 8g of total memory, the agent will effectively have access 1g of memory at all times, assuming there are no other significant applications running on the host. In certain situtations, under high memory load
+where a significant portion of memory is backed by executable code loaded from files on disk, such as when build large dotnet applications, this memory may be evicted before the oom-killer is invoked. This thrashing
+can degrade the performance of the Linux system and the agent so severely that the agent's connection to the server can time out, causing the pipeline to fail.
+
+#### Understanding the Out of Memory Killer
+If a Linux system runs out of memory, it invokes the oom-killer to reclaim memory. The oom-killer chooses a process to sacrifice based on heuristics,
+and adjusted by `oom_score_adj`. Higher scores are more likely to get killed, and range from -1000 to 1000. It is important that the agent process has a lower
+score than the job processes it manages, because if the agent is killed the job effectively dies as well.
+
+The agent can help manage process oom scores. By default, processes that are invoked by the agent will have an oom score of 500. This score can be overriden with
+the `VSTS_JOB_OOMSCOREADJ` environment variable. This value must be between -1000 and 1000 and cannot be less than the `oom_score_adj` of the agent process, which
+is 0 by default. Typically this does not need to change unless there are other processes on your host that need to be more or less likely to be killed.
+
+There are multiple ways to set the agent `oom_score_adj`. When running interactively the score can be set in the shell, and will be inherited by the agent:
+
+```bash
+$ echo $oomScoreAdj > /proc/$$/oom_score_adj
+$ ./run.sh
+```
+
+If the agent is being managed by systemd, the `OOMScoreAdjust` directive can be set in the unit file:
+```
+$ cat /etc/systemd/system/vsts.agent.user.linux-host.service
+[Unit]
+Description=VSTS Agent (user.linux-host)
```

Comment:
A recent PR changed the template to "Azure Pipelines Agent". I can't fix all the "VSTS Agents" out there in the world, but I can stop making new ones :)